### PR TITLE
chore: add linting

### PR DESCRIPTION
**Pull request by bradleyjones**
- chore: add linters and update code to conform
- fix: remove unused constants in v1 handler
- chore: fix linting in http package
- chore: fix linting in the model
- chore: fix linting for adapter package
- chore: fix go imports

### CODE REVIEW

File: pkg/adapter/anchore/adapter.go [hunk 1]
```diff
@@ -354,53 +364,55 @@ func BuildHarborVulnerabilityReport(
 	imageRepository string,
 	imageDigest string,
 	includeDescriptions bool,
-	clientConfig *client.ClientConfig,
+	clientConfig *client.Config,
```
Comment (juanjsebgarcia) [inline pkg/adapter/anchore/adapter.go:367]: We're confident these aren't impacting something?

File: pkg/http/api/v1/handler.go [hunk 2]
```diff
@@ -163,55 +159,56 @@ func (h *APIHandler) CreateScan(res http.ResponseWriter, req *http.Request) {
...
 	var requestedType string
 	var report interface{}
 	var err error
 
-	if numTypes > 1 {
+	switch {
+	case numTypes > 1:
 		log.Info("No support for multiple types per request, yet")
-	} else if numTypes == 0 || requestedTypes[0] == AllMimeTypes || requestedTypes[0] == "" {
+	case numTypes == 0, requestedTypes[0] == AllMimeTypes, requestedTypes[0] == "":
```
Comment (juanjsebgarcia) [inline pkg/http/api/v1/handler.go:177]: Wow nice syntax

File: pkg/http/api/v1/handler.go [hunk 3]
```diff
@@ -6,31 +6,27 @@ import (
...
+	AuthorizationHeader        = "Authorization"
+	BearerTokenPrefix          = "bearer"
+	AllMimeTypes               = "*/*"
+	JSONRequestMimeType        = "application/json"
+	HarborVulnReportv1MimeType = "application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"
+	RawVulnReportMimeType      = "application/vnd.scanner.adapter.vuln.report.raw+json"
+	ScanRequestMimeType        = "application/vnd.scanner.adapter.scan.request+json; version=1.0"
+	ScanResponseMimeType       = "application/vnd.scanner.adapter.scan.response+json; version=1.0"
+	MetadataResponseMimeType   = "application/vnd.scanner.adapter.metadata+json; version=1.0"
+	ErrorResponseMimeType      = "application/vnd.scanner.adapter.error+json; version=1.0"
```
Comment (juanjsebgarcia) [inline pkg/http/api/v1/handler.go:29]: More got deleted here?

File: pkg/adapter/anchore/adapter.go (hunk 1 above)
Comment (bradleyjones) [reply pkg/adapter/anchore/adapter.go:367]: Yeah functionally we're all good and nobody is importing this package for use in other code but if we decide that's a concern when we do the next release we'll mark as a major

File: pkg/http/api/v1/handler.go (hunk 3 above)
Comment (bradleyjones) [reply pkg/http/api/v1/handler.go:29]: Yeah there were a bunch of unused constants so I just removed them to clear up

**Review by juanjsebgarcia — APPROVED** [review]
:shipit:

**Pull request merged**